samples/matter/common/src/bridge/bridge_manager.cpp

```diff
@@ -140,13 +152,13 @@ CHIP_ERROR BridgeManager::HandleWrite(uint16_t index, ClusterId clusterId,
 	VerifyOrReturnError(attributeMetadata && buffer, CHIP_ERROR_INVALID_ARGUMENT);
 	VerifyOrReturnValue(Instance().mDevicesMap.Contains(index), CHIP_ERROR_INTERNAL);
 
-	auto &device = Instance().mDevicesMap[index]->mDevice;
+	auto *device = Instance().mDevicesMap[index].value1;
 	CHIP_ERROR err = device->HandleWrite(clusterId, attributeMetadata->attributeId, buffer);
 
 	/* After updating Matter BridgedDevice state, forward request to the non-Matter device. */
 	if (err == CHIP_NO_ERROR) {
-		return Instance().mDevicesMap[index]->mProvider->UpdateState(clusterId, attributeMetadata->attributeId,
-									     buffer);
+		return Instance().mDevicesMap[index].value2->UpdateState(clusterId, attributeMetadata->attributeId,
```

Comment:
I think, maybe we should concretize this map. I wanted to keep it generic for now to evaluate if it can be reused for `DeviceFactory` but maybe it's pointless. I will either do it, or add some aliases to make the code more readable.

Reply:
I agree that `value1` and `value2` are so confusing here. I think it will be hard to understand by our customers. I know that also, that the map should be as generic as possible, but in this way, it can increase the difficulty to understand the code. Adding some aliases to the code sounds good.

Reply:
Done, I have re-designed the implementation of storage of multiple providers as well as the bridged device removal.